include/RAJA/exec-openmp/scan_openmp.hxx

```diff
@@ -82,6 +82,7 @@ void inclusive_scan_inplace(const omp_parallel_for_exec&,
                             ValueT v)
 {
   using Value = typename std::decay<decltype(*std::declval<Iter&>())>::type;
+  // using Value = typename std::iterator_traits <Iter>::value_type;
```

Comment:
@trws @davidbeckingsale @DavidPoliakoff : Do you have any idea why GCC 5.x would break on either of these when using NVCC? The device-generated code all works as expected and all types get resolved as expected.

Both of these typedefs don't work on my Arch box. Vanilla GCC 5.x, GCC 6.x, and clang 3.8+ (w/ OpenMP support) all work great

```
/usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/include/c++/bits/stl_iterator_base_types.h(158): error: name followed by "::" must be a class or namespace name
          detected during:
            instantiation of class "std::__iterator_traits<_Iterator, void> [with _Iterator=int]"
(163): here
            instantiation of class "std::iterator_traits<_Iterator> [with _Iterator=int]"
```

GCC seems to think that Iter resolves to an int given this compiler output.